pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -12,324 +14,328 @@ import (
 	})
 
 	t.Run("create task: with after", func(t *testing.T) {
-		otherId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
-
-		request := sdk.NewCreateTaskRequest(otherId, sql).WithSchedule(sdk.String("10 MINUTE"))
+		rootTaskId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
 
-		createTaskWithRequest(t, request)
+		err := testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(rootTaskId, sql))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, rootTaskId))
 
-		request = createTaskBasicRequest(t).
-			WithAfter([]sdk.SchemaObjectIdentifier{otherId})
+		err = testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(id, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootTaskId}))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, id))
 
-		task := createTaskWithRequest(t, request)
+		task, err := testClientHelper().Task.Show(t, id)
+		require.NoError(t, err)
 
-		assertTaskWithOptions(t, task, request.GetName(), "", "", "", "", false, "", &otherId)
+		assertTaskWithOptions(t, task, id, "", "", "", "", false, "", &rootTaskId)
 	})
 
-	t.Run("create dag of tasks", func(t *testing.T) {
-		rootId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+	t.Run("create task: with after and finalizer", func(t *testing.T) {
+		rootTaskId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		finalizerId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
 
-		request := sdk.NewCreateTaskRequest(rootId, sql).WithSchedule(sdk.String("10 MINUTE"))
-		root := createTaskWithRequest(t, request)
+		err := testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(rootTaskId, sql))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, rootTaskId))
 
-		require.Empty(t, root.Predecessors)
+		err = testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(id, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootTaskId}))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, id))
 
-		t1Id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		err = testClient(t).Tasks.Create(ctx, sdk.NewCreateTaskRequest(finalizerId, sql).WithFinalize(rootTaskId))
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, finalizerId))
 
-		request = sdk.NewCreateTaskRequest(t1Id, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootId})
-		t1 := createTaskWithRequest(t, request)
+		assertions.AssertThat(t, objectassert.Task(t, rootTaskId).
+			HasTaskRelations(sdk.TaskRelations{
+				Predecessors:  []sdk.SchemaObjectIdentifier{},
+				FinalizerTask: &finalizerId,
+			}),
+		)
+	})
 
-		require.Equal(t, []sdk.SchemaObjectIdentifier{rootId}, t1.Predecessors)
+	t.Run("create dag of tasks", func(t *testing.T) {
+		rootId := testClientHelper().Ids.RandomSchemaObjectIdentifier()
```

Comment:
Nit: it would be great to provide a simple visual representation of the graph.

Reply:
True, added comment above.